app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,62 @@
+class PasswordResetsController < ApplicationController
+  before_action :find_user, :valid_user, :check_expiration,
+    only: [:edit, :update]
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "pass_reset"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "e_not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t(".empty")
```

Comment:
=)) cái này thực ra là cái hiển thị lỗi ở form lúc không tạo được, ví dụ: lúc tạo user lỗi thì mình sẽ có các dòng thông báo password can't be blank chẳng hạn, cái errors.add là để nó hiện thông báo ở trang như vậy còn cái flash chỉ là flash hiển thị báo ko tạo được thôi =))